aws_fuzzy_finder/main.py

```diff
@@ -91,7 +91,11 @@ def entrypoint(use_private_ip, key_path, user, ip_only, no_cache, tunnel, tunnel
 
     chosen_host = choice(fuzzysearch_bash_command, use_ssm)
 
-    if use_ssm or ENV_USE_SSM:
+    if ENV_USE_SSM or use_ssm:
+        if id_only:
```

Comment:
There is a known gap here, if you don't pass --ssm then you can't get to this branch of the code. I looked into using the click library to make dependent variables. It doesn't seem like it's something that works easily out of the box. 

I'm open to making this more robust or adding exception handling but I didn't see a lot of that in this project and decided to just stick with simple.